### DISCUSSION TITLE
Durable subscriptions

### DISCUSSION THREAD

**Discussion by joseftw** [Havret/dotnet-activemq-artemis-client#378] [Ideas]
Hello!

Am I right assuming that this library doesn't support durable subscriptions? ([Example here](https://github.com/apache/activemq-artemis/blob/main/examples/protocols/amqp/dotnet/DurableSubscriptions/Program.cs))

If I'm correct, would you mind if I started working on a PR trying to implement support for it?
If I'm INCORRECT, do you have any examples of how to use durable subscriptions with this library?

Cheers!

**Comment by Havret** [comment]
Hello @joseftw 

Do you specifically need non-shared durable subscriptions? At the moment ArtemisNetClient doesn't support them in the form presented in your example. You can kind of simulate this behavior [using topology manager.](https://github.com/Havret/activemq-artemis-net-reliable-pub-sub)

In general non-shared durable subscriptions are quite problematic as ActiveMQ Artemis doesn't allow you to specify the timeout of the subscription [as the original ActiveMQ does](https://activemq.apache.org/manage-durable-subscribers). If your consumer goes down for whatever reason, and producer keeps pushing new messages, they will keep building up on the queue, and eventually take the broker down with OutOfMemoryException.

I don't know what your use case is, or limitations are (you may not have access to topology management api) but if you think this will be genuinely useful, I'd more than happy to accept your PR.
